Fix 28545: decribe template literals more explicitly (#28631)
mdn/content: files/en-us/learn/javascript/first_steps/strings/index.md

The following will work if you previously defined the variable `string` — try it now:

```js
const badString = string;
console.log(badString);
```

`badString` is now set to have the same value as `string`.

### Single quotes, double quotes, and backticks

In JavaScript, you can choose single quotes (`'`), double quotes (`"`), or backticks (`` ` ``) to wrap your strings in. All of the following will work:

```js-nolint
const single = 'Single quotes';
const double = "Double quotes";
const backtick = `Backtick`;

console.log(single);
console.log(double);
console.log(backtick);
```

You must use the same character for the start and end of a string, or you will get an error:

```js-nolint example-bad
const badQuotes = 'This is not allowed!";
```

> Strings declared using single quotes and strings declared using double quotes are the same, and which you use is down to personal preference — although it is good practice to choose one method and use it consistently in your code.

Reviewer: ```suggestion
Strings declared using single quotes and strings declared using double quotes are the same, and which you use is down to personal preference — although it is good practice to choose one style and use it consistently in your code.
```

> Strings declared using backticks are a special kind of string called a _template literal_. In most ways, template literals are like normal strings, but they have some special properties:

Reviewer: ```suggestion
Strings declared using backticks are a special kind of string called a [_template literal_](/en-US/docs/Web/JavaScript/Reference/Template_literals). In most ways, template literals are like normal strings, but they have some special properties:
```

console.log(joined); // "Hello, how are you?"
```

> Joining strings together like this is called _concatenation_.

Reviewer: TBH I won't call template literals "concatenation"—it's "interpolation" to be exact. But it would be more intractable to change the content below, so it's fine for me.

Template literals respect the line breaks in the source code, so you can write strings that span multiple lines like this:

```js
const newline = `One day you finally knew

> what you had to do, and began,`;

Reviewer: 🤔

Since we use quotes to indicate the start and end of strings, how can we include actual quotes in strings? We know that this won't work:

```js-nolint example-bad
const badQuotes = "She said "I think so!"";
```

One common option is to use one of the other characters to declare the string:

```js-nolint
const goodQuotes1 = 'She said "I think so!"';
const goodQuotes2 = `She said "I'm not going in there!"`;
```

Another option is to _escape_ the problem quoteation mark. Escaping characters means that we do something to them to make sure they are recognized as text, not part of the code. In JavaScript, we do this by putting a backslash just before the character. Try this:

```js-nolint
const bigmouth = 'I\'ve got no right to take my place…';
console.log(bigmouth);
```

You can use the same technique to insert other special characters. See [Escape sequences](/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#escape_sequences) for more details.

## Numbers vs. strings

What happens when we try to concatenate a string and a number? Let's try it in our console:

```js
const name = "Front ";
const number = 242;
console.log(`${name}${number}`); // "Front 242"
```

> You might expect this to return an error, but it works just fine. Trying to represent a string as a number doesn't really make sense, but representing a number as a string does, so the browser converts the number to a string and concatenates the two strings.

Reviewer: ```suggestion
You might expect this to return an error, but it works just fine. How numbers should be displayed as strings is fairly well-defined, so the browser automatically converts the number to a string and concatenates the two strings.
```

(JS can implicitly convert strings to numbers too)

> If you have a numeric variable that you want to convert to a string but not change otherwise, or a string variable that you want to convert to a number but not change otherwise, you can use the following two constructs:

Reviewer: I'm not 100% sure what "not change otherwise" means. I assume you mean "it's still conceptually the same value", but that's somehow implied by "convert to a string".

Author: I don't know, I didn't write this section :). Maybe we could just delete "but not change otherwise"?

Reviewer: Yeah, that would be good.

- The {{jsxref("Number/Number", "Number()")}} function converts anything passed to it into a number, if it can. Try the following:

  ```js
  const myString = "123";
  const myNum = Number(myString);
  console.log(typeof myNum);
  ```

> - Conversely, every number has a method called [`toString()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toString) that converts it to the equivalent string. Try this:

Reviewer: Why not use `String()` instead—that should make them appear more symmetrical.

Template literals respect the line breaks in the source code, so you can write strings that span multiple lines like this:

```js
const newline = `One day you finally knew
what you had to do, and began,`;
console.log(newline);

/*
One day you finally knew
what you had to do, and began,
*/
```

To have the equivalent output using a normal string you'd have to include line break characters (`\n`) in the string:

```js
const newline = "One day you finally knew\nwhat you had to do, and began,";
console.log(newline);

/*
One day you finally knew
what you had to do, and began,
*/
```

See our [Template literals](/en-US/docs/Web/JavaScript/Reference/Template_literals) reference page for more examples and details of advanced features.

## Including quotes in strings

Since we use quotes to indicate the start and end of strings, how can we include actual quotes in strings? We know that this won't work:

```js-nolint example-bad
const badQuotes = "She said "I think so!"";
```

One common option is to use one of the other characters to declare the string:

```js-nolint
const goodQuotes1 = 'She said "I think so!"';
const goodQuotes2 = `She said "I'm not going in there!"`;
```

> Another option is to _escape_ the problem quoteation mark. Escaping characters means that we do something to them to make sure they are recognized as text, not part of the code. In JavaScript, we do this by putting a backslash just before the character. Try this:

Author: ```suggestion
Another option is to _escape_ the problem quotation mark. Escaping characters means that we do something to them to make sure they are recognized as text, not part of the code. In JavaScript, we do this by putting a backslash just before the character. Try this:
```
